src/spyglass/lfp/v1/lfp.py

```diff
@@ -947,3 +949,35 @@ def fetch1_dataframe(self, *attrs, **kwargs):
             filtered_nwb["filtered_data"].data,
             index=pd.Index(filtered_nwb["filtered_data"].timestamps, name="time"),
         )
+
+    def compute_analytic_signal(self, electrode_list, **kwargs):
+        filtered_band = self.fetch_nwb()[0]["filtered_data"]
+        electrode_index = np.isin(filtered_band.electrodes.data[:], electrode_list)
+        analytic_signal_df = pd.DataFrame(
+            hilbert(filtered_band.data[:, electrode_index], axis=0),
+            index=pd.Index(filtered_band.timestamps, name="time"),
+            columns=[f"electrode {e}" for e in electrode_list],
+        )
+        return analytic_signal_df
+
+    def compute_signal_phase(self, electrode_list=[], **kwargs):
+        analytic_signal_df = self.compute_analytic_signal(electrode_list, **kwargs)
+        return pd.DataFrame(
```

Comment:
The for loop is not necessary here. You can just return np.angle(analytic_signal_df)

Reply:
Same question here. the Or is there a way to return a df with the correct column name without using a for loop? @edeno

Reply:
If I understand correctly, the function `compute_analytic_signal` returns a dataframe with the correct electrodes. If you run `np.angle(df) + np.pi`, that will also return a dataframe.

Reply:
Okie I think I figured it out. I can just use np.angle(analytic_signal_df) and then define the columns and index by calling those from the analytic_signal_df. Thanks @dpeg22 @edeno

Reply:
If you run np.angle does it return a numpy array or a pandas dataframe. I just realized that some operations do the former and some operations do the latter, which is where I think the confusion was coming from. If it does the former, this is correct.